File: Lectures/lecture_slides/fibonacci.py
''' select the portions of test code and
    press Alt+3 to comment or
    press Alt+4 to uncomment '''

''' iterative while fibonacci '''
def fib(n):
    counter = 0
    a, b = 1, 0
    while counter < n:
        a, b = a+b, a
        counter = counter + 1
    return b


##for i in range(10):
##    print(fib(i))

''' recursive fibonacci '''
def fib(n):
   if (n == 0): return 0
   elif (n == 1): return 1
   else: return (fib(n - 1) + 
                 fib(n - 2)) 

##for i in range(10):
##    print(fib(i))

File: recitations/rec11/provided answers/Fibonacci.java
public class Fibonacci {

    private static int fib(int n) {
        int i, a = 0, b = 1;
        if (n <= 0) return a;
        if (n == 1) return b;
        for (i=2; i<=n; i++) {
            int temp = a + b;
            a = b;
            b = temp;
        }
        return b;
    }
    
    private static int fib_r(int n) {
      if (n <= 0) return 0;
      if (n==1) return 1;
      return fib_r(n-1) + fib_r(n-2);
    }
    
    public static void main(String [] args) {
        System.out.println(fib(46));
        System.out.println("recursive:" + fib_r(46));
    }
}
